### ISSUE TITLE
Crash shortly after swapping cores from N64 if Glide64mk2 is your graphics plugin

### ISSUE THREAD

**Issue by danne2** [TASVideos/BizHawk#1019]
This occurs in the latest dev build.

If Glide64Mk2 is your graphics plugin then you swap cores, a few seconds after the new core has loaded and the game has started, Bizhawk will hang then proceed to crash with no exception. 

Steps to reproduce:
Set your N64 video plugin to Glide64MK2
Load an N64 rom
Swap to any other core 
Wait around 5-10 seconds
A crash should occur.

**Comment by zeromus** [comment]
Yeah, ive noticed this too. It can happen when closing the emulator too, while debugging. One of those plugins doesnt like being shutdown. I think it happened during the 64bit conversion. It's really hard to debug.

**Comment by danne2** [comment]
One potentially useful tidbit of information is that even in the 1.X branch, there'd be issues on quitting the emulator with Glide64mk2. The process would become a zombie for a good 30 seconds before fully terminating. If you taskkilled it, you'd get an exception about failing to read memory if I remember correctly

**Comment by danne2** [comment]
Also I am able to replicate this 100% of the time using Glide64mk2 but if I have GlideN64 set, it doesn't occur.

**Comment by YoshiRulz** [comment]
Reproducible on BizHawk 2.3, OS: Win10 Home 1709. Loading a NES rom causes a crash, quitting or loading another N64 rom does not.

**Comment by zeromus** [comment]
This actually happens when simply closing the emulator after loading a rom with glide64mk2 (as this closes the core, and the problem is during closing the core)
After some investigation I disovered commenting out mupen64pluscoreapi.cs `FreeLibrary(CoreDll);` would fix it. Therefore we have the possibility for a tedious but feasible process of locating the problem by systematically deleting chunks of the plugin and testing repeatedly until we find the logic that's to blame.

**Comment by YoshiRulz** [comment]
Related to #784?

**Comment by zeromus** [comment]
probably
